clients/src/test/java/org/apache/kafka/clients/consumer/internals/RequestStateTest.java

```diff
@@ -48,4 +50,51 @@ public void testRequestStateSimple() {
         state.reset();
         assertTrue(state.canSendRequest(200));
     }
+
+    @Test
+    public void testTrackInflightOnSuccessfulAttempt() {
+        testTrackInflight(RequestState::onSuccessfulAttempt);
+    }
+
+    @Test
+    public void testTrackInflightOnFailedAttempt() {
+        testTrackInflight(RequestState::onFailedAttempt);
+    }
+
+    /**
+     * In some cases, the network layer is <em>very</em> fast and can send out a second request within the same
+     * millisecond timestamp as receiving the first response.
+     *
+     * <p/>
+     *
+     * The previous logic for tracking inflight status used timestamps: if the timestamp from the last received
+     * response was <em>less</em> than the timestamp from the last sent request, we'd interpret that as having an
+     * inflight request. However, this approach would incorrectly return <code>false</code> from
+     * {@link RequestState#requestInFlight()} if the two timestamps were <em>equal</em>.
+     */
```

Comment:
nit: Do we think it's helpful to explain so much about a "previous logic" that was wrong here? I find it can be confusing for others. Also here we're actually just testing the approach of checking inflights based on a flag (the why we chose that approach makes more sense to me in the `requestInflight()` func itself)

Reply:
yeah I also noticed this. I think this parapraph fits best in the commit message / PR description.

Reply:
I figure the existing PR description covers the basics. I'll just remove the test comment wholesale.